PIL/GifImagePlugin.py

```diff
@@ -434,18 +434,16 @@ def _get_local_header(fp, im, offset, flags):
         # optimize the block away if transparent color is not used
         transparent_color_exists = True
 
-        if _get_optimize(im, im.encoderinfo):
-            used_palette_colors = _get_used_palette_colors(im)
-
+        used_palette_colors = _get_optimize(im, im.encoderinfo)
+        if used_palette_colors is not None:
             # adjust the transparency index after optimize
-            if len(used_palette_colors) < 256:
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+            for i, palette_color in enumerate(used_palette_colors):
+                if palette_color == transparency:
+                    transparency = i
+                    transparent_color_exists = True
+                    break
+                else:
```

Comment:
This should be `else` clause for `for` loop (ie when break isn't called). Or, better, `transparent_color_exists = False` should become before the loop.